Foundation/src/Environment_WIN32U.cpp

```diff
@@ -209,31 +209,65 @@ void EnvironmentImpl::nodeIdImpl(NodeId& id)
 	// Make an initial call to GetAdaptersInfo to get
 	// the necessary size into len
 	DWORD rc = GetAdaptersInfo(pAdapterInfo, &len);
-	if (rc == ERROR_BUFFER_OVERFLOW)
-	{
-		delete [] reinterpret_cast<char*>(pAdapterInfo);
-		pAdapterInfo = reinterpret_cast<IP_ADAPTER_INFO*>(new char[len]);
-	}
-	else if (rc != ERROR_SUCCESS)
-	{
-		delete[] reinterpret_cast<char*>(pAdapterInfo);
-		throw SystemException("cannot get network adapter list");
-	}
-	if (GetAdaptersInfo(pAdapterInfo, &len) == NO_ERROR)
+
+	try
```

Comment:
Still consider doing something like: (not tested)

```c++
	ULONG len = sizeof(IP_ADAPTER_INFO);

	std::unique_ptr<char[]> buffer = std::make_unique<char[]>(len);
	pAdapterInfo = reinterpret_cast<IP_ADAPTER_INFO*>(buffer.get());

	// Make an initial call to GetAdaptersInfo to get
	// the necessary size into len
	DWORD rc = GetAdaptersInfo(pAdapterInfo, &len);
	if (rc == ERROR_BUFFER_OVERFLOW)
	{
		buffer = std::make_unique<char[]>(len);
		pAdapterInfo = reinterpret_cast<IP_ADAPTER_INFO*>(buffer.get());
	}
```

That way you do not need try catch at all.

Reply:
I was trying to avoid new things for this PR. I only see `std::make_unique` in a couple of the newer Poco modules. The bigger issue here is the static alias violations jumping between char* and IP_ADAPTER_INFO*. I think I have a fix for that part.